### PR TITLE
KNE-26: Store jsapi auth token in redis

### PR DESCRIPTION
**Pull request by rossnelson**
All kne processes currently request a auth token on init. Storing the token in
redis allows us to pair down the amount of requests that made for server side
authentication across all node processes and instances.

- Adds get and set functions inside getToken
- If token expires, remove redis stored token before calling getToken again.

### CODE REVIEW

File: submodules/kneJsapi.js [hunk 1]
```diff
@@ -5,6 +5,8 @@
 var _ = require('lodash'),
     Q = require('q'),
     rest = require('restler'),
+    kneRedis = null,
+    rediClient = null,
```
Comment (kn-rjoswick) [inline submodules/kneJsapi.js]: I bet you mean redisClient here.

File: submodules/kneJsapi.js [hunk 2]
```diff
@@ -226,15 +232,54 @@ KneJsapi.prototype.wrapper = function (fn, options) {
...
-    }));
+    const self = this;
+
+    return this.getCurrent()
+    .then(res => {
+        if (res) { return res; }
+
+        const postMeth = rest.post.bind(self);
+        const oauthUrl = `${self.engine.getConfig('kne-jsapi.url')}/oauth/token`;
+        const oauthReq = postMeth(oauthUrl, {
```
Comment (kn-rjoswick) [inline submodules/kneJsapi.js]: postMeth()?

File: submodules/kneJsapi.js [hunk 3]
```diff
@@ -254,11 +299,13 @@ KneJsapi.prototype.authenticationMiddleware = function (socket, next) {
...
             accessToken: socket.handshake.query.token
-        }), { 'allowTokenRefresh': allowTokenRefresh }).then(function (res) {
+        }), { 'allowTokenRefresh': allowTokenRefresh })
+        .then(function (res) {
             res.result.token = socket.handshake.query.token;
             socket.userData = res.result;
-            next();
-        }, function (err) {
+            next(socket);
+        })
```
Comment (jgpacheco) [inline submodules/kneJsapi.js]: If all goes well `next` should be called without errors `next();`

Comment (rossnelson) [inline submodules/kneJsapi.js]: yep, we can remove that.

**Pull request merged**